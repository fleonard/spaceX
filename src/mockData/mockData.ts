const dragon = {
  id: "dragon1",
  name: "Dragon 1",
  type: "capsule",
  active: true,
  crew_capacity: 0,
  sidewall_angle_deg: 15,
  orbit_duration_yr: 2,
  dry_mass_kg: 4200,
  dry_mass_lb: 9300,
  first_flight: "2010-12-8",
  flickr_images: [
    "https://imgur.com/DaCfMsj.jpg",
    "https://imgur.com/azYafd8.jpg",
  ],
  heat_shield: {
    material: "PICA-X",
    size_meters: 3.6,
    temp_degrees: 3000,
    dev_partner: "NASA",
  },
  thrusters: [
    {
      type: "Draco",
      amount: 18,
      pods: 4,
      isp: 300,
      fuel_1: "nitrogen tetroxide",
      fuel_2: "monomethylhydrazine",
      thrust: {
        kN: 0.4,
        lbf: 90,
      },
    },
  ],
  launch_payload_mass: {
    kg: 6000,
    lb: 13228,
  },
  launch_payload_vol: {
    cubic_meters: 25,
    cubic_feet: 883,
  },
  return_payload_mass: {
    kg: 3000,
    lb: 6614,
  },
  return_payload_vol: {
    cubic_meters: 11,
    cubic_feet: 388,
  },
  pressurized_capsule: {
    payload_volume: {
      cubic_meters: 11,
      cubic_feet: 388,
    },
  },
  trunk: {
    trunk_volume: {
      cubic_meters: 14,
      cubic_feet: 494,
    },
    cargo: {
      solar_array: 2,
      unpressurized_cargo: true,
    },
  },
  height_w_trunk: {
    meters: 7.2,
    feet: 23.6,
  },
  diameter: {
    meters: 3.7,
    feet: 12,
  },
  wikipedia: "https://en.wikipedia.org/wiki/SpaceX_Dragon",
  description:
    "Dragon is a reusable spacecraft developed by SpaceX, an American private space transportation company based in Hawthorne, California. Dragon is launched into space by the SpaceX Falcon 9 two-stage-to-orbit launch vehicle. The Dragon spacecraft was originally designed for human travel, but so far has only been used to deliver cargo to the International Space Station (ISS).",
};

const rocket = {
  id: 1,
  active: false,
  stages: 2,
  boosters: 0,
  cost_per_launch: 6700000,
  success_rate_pct: 40,
  first_flight: "2006-03-24",
  country: "Republic of the Marshall Islands",
  company: "SpaceX",
  height: {
    meters: 22.25,
    feet: 73,
  },
  diameter: {
    meters: 1.68,
    feet: 5.5,
  },
  mass: {
    kg: 30146,
    lb: 66460,
  },
  payload_weights: [
    {
      id: "leo",
      name: "Low Earth Orbit",
      kg: 450,
      lb: 992,
    },
  ],
  first_stage: {
    reusable: false,
    engines: 1,
    fuel_amount_tons: 44.3,
    burn_time_sec: 169,
    thrust_sea_level: {
      kN: 420,
      lbf: 94000,
    },
    thrust_vacuum: {
      kN: 480,
      lbf: 110000,
    },
  },
  second_stage: {
    reusable: false,
    engines: 1,
    fuel_amount_tons: 3.38,
    burn_time_sec: 378,
    thrust: {
      kN: 31,
      lbf: 7000,
    },
    payloads: {
      option_1: "composite fairing",
      composite_fairing: {
        height: {
          meters: 3.5,
          feet: 11.5,
        },
        diameter: {
          meters: 1.5,
          feet: 4.9,
        },
      },
    },
  },
  engines: {
    number: 1,
    type: "merlin",
    version: "1C",
    layout: "single",
    engine_loss_max: 0,
    propellant_1: "liquid oxygen",
    propellant_2: "RP-1 kerosene",
    thrust_sea_level: {
      kN: 420,
      lbf: 94000,
    },
    thrust_vacuum: {
      kN: 480,
      lbf: 110000,
    },
    thrust_to_weight: 96,
  },
  landing_legs: {
    number: 0,
    material: null,
  },
  wikipedia: "https://en.wikipedia.org/wiki/Falcon_1",
  description:
    "The Falcon 1 was an expendable launch system privately developed and manufactured by SpaceX during 2006-2009. On 28 September 2008, Falcon 1 became the first privately-developed liquid-fuel launch vehicle to go into orbit around the Earth.",
  rocket_id: "falcon1",
  rocket_name: "Falcon 1",
  rocket_type: "rocket",
  flickr_images: [
    "https://imgur.com/DaCfMsj.jpg",
    "https://imgur.com/azYafd8.jpg",
  ],
};

export { rocket, dragon };
