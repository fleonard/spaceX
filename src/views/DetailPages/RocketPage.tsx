import React from "react";
import { Redirect, useLocation } from "react-router-dom";

import { TableBody, TableCell, TableRow } from "@material-ui/core";

import { Rocket } from "../../types/apiTypes";

import { ColumnContainer } from "../../globalStyles";

import {
  StyledSection,
  ImageContainer,
  InfoContainer,
  StyledTable,
} from "./StyledDetailPages";

interface LocationState {
  item: Rocket;
}

const RocketPage: React.FC = () => {
  const location = useLocation();

  if (location.state === undefined) {
    return <Redirect to="/rockets" />;
  }

  const rocket = (location.state as LocationState).item;

  const {
    company,
    cost_per_launch,
    country,
    description,
    diameter,
    engines,
    first_stage,
    flickr_images,
    height,
    mass,
    rocket_name,
    second_stage,
    wikipedia,
  } = rocket;

  return (
    <ColumnContainer>
      <StyledSection data-testid="informations">
        <ImageContainer>
          <img src={flickr_images[0]} alt={rocket_name} />
        </ImageContainer>
        <InfoContainer>
          <h2>{rocket_name}</h2>
          <p>{description}</p>
          <h3>Additional Informations</h3>
          <StyledTable aria-label="simple table">
            <TableBody>
              <TableRow>
                <TableCell component="th" scope="row">
                  Company
                </TableCell>
                <TableCell align="right">{company}</TableCell>
              </TableRow>
              <TableRow>
                <TableCell component="th" scope="row">
                  Country
                </TableCell>
                <TableCell align="right">{country}</TableCell>
              </TableRow>
              <TableRow>
                <TableCell component="th" scope="row">
                  Cost Per Launch
                </TableCell>
                <TableCell align="right">{cost_per_launch}</TableCell>
              </TableRow>
            </TableBody>
          </StyledTable>
          <a href={wikipedia} target="_blank" rel="noopener noreferrer">
            Checkout wikipidia page for more.
          </a>
        </InfoContainer>
      </StyledSection>
      <StyledSection data-testid="overview">
        <InfoContainer>
          <h3>Rocket</h3>
          <h2>Overview</h2>
          <StyledTable aria-label="simple table">
            <TableBody>
              <TableRow>
                <TableCell component="th" scope="row">
                  Diameter
                </TableCell>
                <TableCell align="right">
                  {diameter.meters} m<span> / {diameter.feet} ft</span>
                </TableCell>
              </TableRow>
              <TableRow>
                <TableCell component="th" scope="row">
                  Height
                </TableCell>
                <TableCell align="right">
                  {height.meters} m<span> / {height.feet} ft</span>
                </TableCell>
              </TableRow>
              <TableRow>
                <TableCell component="th" scope="row">
                  Mass
                </TableCell>
                <TableCell align="right">
                  {mass.kg} kg<span> / {mass.lb} lb</span>
                </TableCell>
              </TableRow>
            </TableBody>
          </StyledTable>
        </InfoContainer>
        {flickr_images[1].length ? (
          <ImageContainer>
            <img src={flickr_images[1]} alt={rocket_name} />
          </ImageContainer>
        ) : (
          ""
        )}
      </StyledSection>
      <StyledSection data-testid="engines">
        <InfoContainer>
          <h3>Engines</h3>
          <h2>Overview</h2>
          <StyledTable aria-label="simple table">
            <TableBody>
              <TableRow>
                <TableCell component="th" scope="row">
                  Number
                </TableCell>
                <TableCell align="right">{engines.number}</TableCell>
              </TableRow>
              <TableRow>
                <TableCell component="th" scope="row">
                  Type
                </TableCell>
                <TableCell align="right">{engines.type}</TableCell>
              </TableRow>
              <TableRow>
                <TableCell component="th" scope="row">
                  Version
                </TableCell>
                <TableCell align="right">{engines.version}</TableCell>
              </TableRow>
              <TableRow>
                <TableCell component="th" scope="row">
                  Layout
                </TableCell>
                <TableCell align="right">{engines.layout}</TableCell>
              </TableRow>
              <TableRow>
                <TableCell component="th" scope="row">
                  Propellant 1
                </TableCell>
                <TableCell align="right">{engines.propellant_1}</TableCell>
              </TableRow>
              <TableRow>
                <TableCell component="th" scope="row">
                  Propellant 2
                </TableCell>
                <TableCell align="right">{engines.propellant_2}</TableCell>
              </TableRow>
              <TableRow>
                <TableCell component="th" scope="row">
                  Thrust sea level
                </TableCell>
                <TableCell align="right">
                  {engines.thrust_sea_level.kN} kN
                  <span> / {engines.thrust_sea_level.lbf} lbf</span>
                </TableCell>
              </TableRow>
              <TableRow>
                <TableCell component="th" scope="row">
                  Thrust vacuum
                </TableCell>
                <TableCell align="right">
                  {engines.thrust_vacuum.kN} kN
                  <span> / {engines.thrust_vacuum.lbf} lbf</span>
                </TableCell>
              </TableRow>
              <TableRow>
                <TableCell component="th" scope="row">
                  Thrust to weigth
                </TableCell>
                <TableCell align="right">{engines.thrust_to_weight}</TableCell>
              </TableRow>
            </TableBody>
          </StyledTable>
        </InfoContainer>
      </StyledSection>
      <StyledSection data-testid="first-stage">
        <InfoContainer>
          <h3>First Stage</h3>
          <h2>Overview</h2>
          <StyledTable aria-label="simple table">
            <TableBody>
              <TableRow>
                <TableCell component="th" scope="row">
                  Burn Time
                </TableCell>
                <TableCell align="right">
                  {first_stage.burn_time_sec} sec
                </TableCell>
              </TableRow>
              <TableRow>
                <TableCell component="th" scope="row">
                  Engines
                </TableCell>
                <TableCell align="right">{first_stage.engines}</TableCell>
              </TableRow>
              <TableRow>
                <TableCell component="th" scope="row">
                  Fuel Amount
                </TableCell>
                <TableCell align="right">
                  {first_stage.fuel_amount_tons} tons
                </TableCell>
              </TableRow>
              <TableRow>
                <TableCell component="th" scope="row">
                  Reusable
                </TableCell>
                <TableCell align="right">
                  {first_stage.reusable ? "Yes" : "No"}
                </TableCell>
              </TableRow>
              <TableRow>
                <TableCell component="th" scope="row">
                  Thrust sea level
                </TableCell>
                <TableCell align="right">
                  {first_stage.thrust_sea_level.kN} kN
                  <span> / {first_stage.thrust_sea_level.lbf} lbf</span>
                </TableCell>
              </TableRow>
              <TableRow>
                <TableCell component="th" scope="row">
                  Thrust vacuum
                </TableCell>
                <TableCell align="right">
                  {first_stage.thrust_vacuum.kN} kN
                  <span> / {first_stage.thrust_vacuum.lbf} lbf</span>
                </TableCell>
              </TableRow>
            </TableBody>
          </StyledTable>
        </InfoContainer>
      </StyledSection>
      <StyledSection data-testid="second-stage">
        <InfoContainer>
          <h3>Second Stage</h3>
          <h2>Overview</h2>
          <StyledTable aria-label="simple table">
            <TableBody>
              <TableRow>
                <TableCell component="th" scope="row">
                  Burn Time
                </TableCell>
                <TableCell align="right">
                  {second_stage.burn_time_sec} sec
                </TableCell>
              </TableRow>
              <TableRow>
                <TableCell component="th" scope="row">
                  Engines
                </TableCell>
                <TableCell align="right">{second_stage.engines}</TableCell>
              </TableRow>
              <TableRow>
                <TableCell component="th" scope="row">
                  Fuel Amount
                </TableCell>
                <TableCell align="right">
                  {second_stage.fuel_amount_tons} tons
                </TableCell>
              </TableRow>
              <TableRow>
                <TableCell component="th" scope="row">
                  Reusable
                </TableCell>
                <TableCell align="right">
                  {second_stage.reusable ? "Yes" : "No"}
                </TableCell>
              </TableRow>
              <TableRow>
                <TableCell component="th" scope="row">
                  Thrust
                </TableCell>
                <TableCell align="right">
                  {second_stage.thrust.kN} kN
                  <span> / {second_stage.thrust.lbf} lbf</span>
                </TableCell>
              </TableRow>
            </TableBody>
          </StyledTable>
        </InfoContainer>
      </StyledSection>
    </ColumnContainer>
  );
};

export default RocketPage;
