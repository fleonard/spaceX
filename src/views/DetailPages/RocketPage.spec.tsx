import React from "react";
import { render } from "@testing-library/react";
import { Router } from "react-router-dom";
import { createMemoryHistory } from "history";
import "jest-styled-components";

import { rocket } from "../../mockData/mockData";

import RocketPage from "./RocketPage";

describe("RocketPage Component", () => {
  it("renders correctly", () => {
    const history = createMemoryHistory();
    const state = { item: rocket };
    history.push("/rockets/falcon1", state);

    const { container, getByText, getByTestId } = render(
      <Router history={history}>
        <RocketPage />
      </Router>,
    );
    expect(container.firstChild).toMatchSnapshot();
    expect(getByText("Falcon 1")).toBeDefined();
    expect(getByTestId("informations")).toBeDefined();
    expect(getByTestId("overview")).toBeDefined();
    expect(getByTestId("engines")).toBeDefined();
    expect(getByTestId("first-stage")).toBeDefined();
    expect(getByTestId("second-stage")).toBeDefined();
  });
});
