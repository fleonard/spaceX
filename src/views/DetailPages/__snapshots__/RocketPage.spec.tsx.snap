// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RocketPage Component renders correctly 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  padding: 15px 20px;
}

.c1 {
  border-bottom: 1px solid #dadce0;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  padding: 25px 0;
}

.c1:first-child {
  padding-top: 25px;
}

.c1:last-child {
  border: 0;
}

.c2 {
  display: block;
  padding-bottom: 20px;
  text-align: center;
  width: 100%;
}

.c2 img {
  height: auto;
  max-width: 600px;
  width: 100%;
}

.c3 {
  padding-bottom: 20px;
  width: 100%;
}

.c3 a:hover {
  -webkit-text-decoration: underline;
  text-decoration: underline;
}

.c3 table span {
  color: #868686;
}

.c4.c4 {
  min-width: 600px;
  margin-bottom: 20px;
}

@media (min-width:1000px) {
  .c1 {
    -webkit-align-items: flex-start;
    -webkit-box-align: flex-start;
    -ms-flex-align: flex-start;
    align-items: flex-start;
    -webkit-flex-direction: row;
    -ms-flex-direction: row;
    flex-direction: row;
    padding: 80px 0;
  }
}

@media (min-width:1000px) {
  .c2 {
    padding-bottom: 0;
    padding-right: 20px;
  }
}

@media (min-width:1000px) {
  .c3 {
    padding-bottom: 0px;
    padding-right: 20px;
  }
}

<div
  class="c0"
>
  <section
    class="c1"
    data-testid="informations"
  >
    <div
      class="c2"
    >
      <img
        alt="Falcon 1"
        src="https://imgur.com/DaCfMsj.jpg"
      />
    </div>
    <div
      class="c3"
    >
      <h2>
        Falcon 1
      </h2>
      <p>
        The Falcon 1 was an expendable launch system privately developed and manufactured by SpaceX during 2006-2009. On 28 September 2008, Falcon 1 became the first privately-developed liquid-fuel launch vehicle to go into orbit around the Earth.
      </p>
      <h3>
        Additional Informations
      </h3>
      <table
        aria-label="simple table"
        class="MuiTable-root c4"
      >
        <tbody
          class="MuiTableBody-root"
        >
          <tr
            class="MuiTableRow-root"
          >
            <th
              class="MuiTableCell-root MuiTableCell-body"
              role="cell"
              scope="row"
            >
              Company
            </th>
            <td
              class="MuiTableCell-root MuiTableCell-body MuiTableCell-alignRight"
            >
              SpaceX
            </td>
          </tr>
          <tr
            class="MuiTableRow-root"
          >
            <th
              class="MuiTableCell-root MuiTableCell-body"
              role="cell"
              scope="row"
            >
              Country
            </th>
            <td
              class="MuiTableCell-root MuiTableCell-body MuiTableCell-alignRight"
            >
              Republic of the Marshall Islands
            </td>
          </tr>
          <tr
            class="MuiTableRow-root"
          >
            <th
              class="MuiTableCell-root MuiTableCell-body"
              role="cell"
              scope="row"
            >
              Cost Per Launch
            </th>
            <td
              class="MuiTableCell-root MuiTableCell-body MuiTableCell-alignRight"
            >
              6700000
            </td>
          </tr>
        </tbody>
      </table>
      <a
        href="https://en.wikipedia.org/wiki/Falcon_1"
        rel="noopener noreferrer"
        target="_blank"
      >
        Checkout wikipidia page for more.
      </a>
    </div>
  </section>
  <section
    class="c1"
    data-testid="overview"
  >
    <div
      class="c3"
    >
      <h3>
        Rocket
      </h3>
      <h2>
        Overview
      </h2>
      <table
        aria-label="simple table"
        class="MuiTable-root c4"
      >
        <tbody
          class="MuiTableBody-root"
        >
          <tr
            class="MuiTableRow-root"
          >
            <th
              class="MuiTableCell-root MuiTableCell-body"
              role="cell"
              scope="row"
            >
              Diameter
            </th>
            <td
              class="MuiTableCell-root MuiTableCell-body MuiTableCell-alignRight"
            >
              1.68
               m
              <span>
                 / 
                5.5
                 ft
              </span>
            </td>
          </tr>
          <tr
            class="MuiTableRow-root"
          >
            <th
              class="MuiTableCell-root MuiTableCell-body"
              role="cell"
              scope="row"
            >
              Height
            </th>
            <td
              class="MuiTableCell-root MuiTableCell-body MuiTableCell-alignRight"
            >
              22.25
               m
              <span>
                 / 
                73
                 ft
              </span>
            </td>
          </tr>
          <tr
            class="MuiTableRow-root"
          >
            <th
              class="MuiTableCell-root MuiTableCell-body"
              role="cell"
              scope="row"
            >
              Mass
            </th>
            <td
              class="MuiTableCell-root MuiTableCell-body MuiTableCell-alignRight"
            >
              30146
               kg
              <span>
                 / 
                66460
                 lb
              </span>
            </td>
          </tr>
        </tbody>
      </table>
    </div>
    <div
      class="c2"
    >
      <img
        alt="Falcon 1"
        src="https://imgur.com/azYafd8.jpg"
      />
    </div>
  </section>
  <section
    class="c1"
    data-testid="engines"
  >
    <div
      class="c3"
    >
      <h3>
        Engines
      </h3>
      <h2>
        Overview
      </h2>
      <table
        aria-label="simple table"
        class="MuiTable-root c4"
      >
        <tbody
          class="MuiTableBody-root"
        >
          <tr
            class="MuiTableRow-root"
          >
            <th
              class="MuiTableCell-root MuiTableCell-body"
              role="cell"
              scope="row"
            >
              Number
            </th>
            <td
              class="MuiTableCell-root MuiTableCell-body MuiTableCell-alignRight"
            >
              1
            </td>
          </tr>
          <tr
            class="MuiTableRow-root"
          >
            <th
              class="MuiTableCell-root MuiTableCell-body"
              role="cell"
              scope="row"
            >
              Type
            </th>
            <td
              class="MuiTableCell-root MuiTableCell-body MuiTableCell-alignRight"
            >
              merlin
            </td>
          </tr>
          <tr
            class="MuiTableRow-root"
          >
            <th
              class="MuiTableCell-root MuiTableCell-body"
              role="cell"
              scope="row"
            >
              Version
            </th>
            <td
              class="MuiTableCell-root MuiTableCell-body MuiTableCell-alignRight"
            >
              1C
            </td>
          </tr>
          <tr
            class="MuiTableRow-root"
          >
            <th
              class="MuiTableCell-root MuiTableCell-body"
              role="cell"
              scope="row"
            >
              Layout
            </th>
            <td
              class="MuiTableCell-root MuiTableCell-body MuiTableCell-alignRight"
            >
              single
            </td>
          </tr>
          <tr
            class="MuiTableRow-root"
          >
            <th
              class="MuiTableCell-root MuiTableCell-body"
              role="cell"
              scope="row"
            >
              Propellant 1
            </th>
            <td
              class="MuiTableCell-root MuiTableCell-body MuiTableCell-alignRight"
            >
              liquid oxygen
            </td>
          </tr>
          <tr
            class="MuiTableRow-root"
          >
            <th
              class="MuiTableCell-root MuiTableCell-body"
              role="cell"
              scope="row"
            >
              Propellant 2
            </th>
            <td
              class="MuiTableCell-root MuiTableCell-body MuiTableCell-alignRight"
            >
              RP-1 kerosene
            </td>
          </tr>
          <tr
            class="MuiTableRow-root"
          >
            <th
              class="MuiTableCell-root MuiTableCell-body"
              role="cell"
              scope="row"
            >
              Thrust sea level
            </th>
            <td
              class="MuiTableCell-root MuiTableCell-body MuiTableCell-alignRight"
            >
              420
               kN
              <span>
                 / 
                94000
                 lbf
              </span>
            </td>
          </tr>
          <tr
            class="MuiTableRow-root"
          >
            <th
              class="MuiTableCell-root MuiTableCell-body"
              role="cell"
              scope="row"
            >
              Thrust vacuum
            </th>
            <td
              class="MuiTableCell-root MuiTableCell-body MuiTableCell-alignRight"
            >
              480
               kN
              <span>
                 / 
                110000
                 lbf
              </span>
            </td>
          </tr>
          <tr
            class="MuiTableRow-root"
          >
            <th
              class="MuiTableCell-root MuiTableCell-body"
              role="cell"
              scope="row"
            >
              Thrust to weigth
            </th>
            <td
              class="MuiTableCell-root MuiTableCell-body MuiTableCell-alignRight"
            >
              96
            </td>
          </tr>
        </tbody>
      </table>
    </div>
  </section>
  <section
    class="c1"
    data-testid="first-stage"
  >
    <div
      class="c3"
    >
      <h3>
        First Stage
      </h3>
      <h2>
        Overview
      </h2>
      <table
        aria-label="simple table"
        class="MuiTable-root c4"
      >
        <tbody
          class="MuiTableBody-root"
        >
          <tr
            class="MuiTableRow-root"
          >
            <th
              class="MuiTableCell-root MuiTableCell-body"
              role="cell"
              scope="row"
            >
              Burn Time
            </th>
            <td
              class="MuiTableCell-root MuiTableCell-body MuiTableCell-alignRight"
            >
              169
               sec
            </td>
          </tr>
          <tr
            class="MuiTableRow-root"
          >
            <th
              class="MuiTableCell-root MuiTableCell-body"
              role="cell"
              scope="row"
            >
              Engines
            </th>
            <td
              class="MuiTableCell-root MuiTableCell-body MuiTableCell-alignRight"
            >
              1
            </td>
          </tr>
          <tr
            class="MuiTableRow-root"
          >
            <th
              class="MuiTableCell-root MuiTableCell-body"
              role="cell"
              scope="row"
            >
              Fuel Amount
            </th>
            <td
              class="MuiTableCell-root MuiTableCell-body MuiTableCell-alignRight"
            >
              44.3
               tons
            </td>
          </tr>
          <tr
            class="MuiTableRow-root"
          >
            <th
              class="MuiTableCell-root MuiTableCell-body"
              role="cell"
              scope="row"
            >
              Reusable
            </th>
            <td
              class="MuiTableCell-root MuiTableCell-body MuiTableCell-alignRight"
            >
              No
            </td>
          </tr>
          <tr
            class="MuiTableRow-root"
          >
            <th
              class="MuiTableCell-root MuiTableCell-body"
              role="cell"
              scope="row"
            >
              Thrust sea level
            </th>
            <td
              class="MuiTableCell-root MuiTableCell-body MuiTableCell-alignRight"
            >
              420
               kN
              <span>
                 / 
                94000
                 lbf
              </span>
            </td>
          </tr>
          <tr
            class="MuiTableRow-root"
          >
            <th
              class="MuiTableCell-root MuiTableCell-body"
              role="cell"
              scope="row"
            >
              Thrust vacuum
            </th>
            <td
              class="MuiTableCell-root MuiTableCell-body MuiTableCell-alignRight"
            >
              480
               kN
              <span>
                 / 
                110000
                 lbf
              </span>
            </td>
          </tr>
        </tbody>
      </table>
    </div>
  </section>
  <section
    class="c1"
    data-testid="second-stage"
  >
    <div
      class="c3"
    >
      <h3>
        Second Stage
      </h3>
      <h2>
        Overview
      </h2>
      <table
        aria-label="simple table"
        class="MuiTable-root c4"
      >
        <tbody
          class="MuiTableBody-root"
        >
          <tr
            class="MuiTableRow-root"
          >
            <th
              class="MuiTableCell-root MuiTableCell-body"
              role="cell"
              scope="row"
            >
              Burn Time
            </th>
            <td
              class="MuiTableCell-root MuiTableCell-body MuiTableCell-alignRight"
            >
              378
               sec
            </td>
          </tr>
          <tr
            class="MuiTableRow-root"
          >
            <th
              class="MuiTableCell-root MuiTableCell-body"
              role="cell"
              scope="row"
            >
              Engines
            </th>
            <td
              class="MuiTableCell-root MuiTableCell-body MuiTableCell-alignRight"
            >
              1
            </td>
          </tr>
          <tr
            class="MuiTableRow-root"
          >
            <th
              class="MuiTableCell-root MuiTableCell-body"
              role="cell"
              scope="row"
            >
              Fuel Amount
            </th>
            <td
              class="MuiTableCell-root MuiTableCell-body MuiTableCell-alignRight"
            >
              3.38
               tons
            </td>
          </tr>
          <tr
            class="MuiTableRow-root"
          >
            <th
              class="MuiTableCell-root MuiTableCell-body"
              role="cell"
              scope="row"
            >
              Reusable
            </th>
            <td
              class="MuiTableCell-root MuiTableCell-body MuiTableCell-alignRight"
            >
              No
            </td>
          </tr>
          <tr
            class="MuiTableRow-root"
          >
            <th
              class="MuiTableCell-root MuiTableCell-body"
              role="cell"
              scope="row"
            >
              Thrust
            </th>
            <td
              class="MuiTableCell-root MuiTableCell-body MuiTableCell-alignRight"
            >
              31
               kN
              <span>
                 / 
                7000
                 lbf
              </span>
            </td>
          </tr>
        </tbody>
      </table>
    </div>
  </section>
</div>
`;
